src/webaudio.js

```diff
@@ -128,6 +128,30 @@ WaveSurfer.WebAudio = {
         this.scriptNode.onaudioprocess = null;
     },
 
+    createChannelNodes: function () {
+        var channels = this.buffer.numberOfChannels;
+
+        this.splitter = this.ac.createChannelSplitter(channels);
+        this.merger = this.ac.createChannelMerger(channels);
+
+        this.setChannel(this.params.channel);
+
+        this.analyser.disconnect();
+        this.analyser.connect(this.splitter);
+
+        this.merger.connect(this.gainNode);
+    },
+
+    setChannel: function (channel) {
+        var channels = this.buffer.numberOfChannels;
+
+        this.splitter.disconnect();
+
+        for (var c = 0; c < channels; c++) {
+            this.splitter.connect(this.merger, channel === -1 ? c : channel, c);
+        }
```

Comment:
I tried to omit the third parameter also, but the output of splitter is only connected to the first channel of the merger (tested on newest Chrome/Firefox).

Although the last two parameters are marked as optional in the document of [AudioNode.connect](https://webaudio.github.io/web-audio-api/#widl-AudioNode-connect-AudioNode-AudioNode-destination-unsigned-long-output-unsigned-long-input), the default behaviour is not described.